bin/image_insert.py

```diff
@@ -0,0 +1,180 @@
+import numpy as np
+import sqlalchemy as sa
+from tqdm import tqdm
+import sys
+import argparse
+
+def insert_image(person, image_file, image_type):
+    engine = sa.create_engine('sqlite:///test.db', echo=False)
+    metadata = sa.MetaData()
+
+    person_tbl = sa.Table('person', metadata, autoload=True, autoload_with=engine)
+    img_tbl    = sa.Table('img'   , metadata, autoload=True, autoload_with=engine)
+    points_tbl = sa.Table('points', metadata, autoload=True, autoload_with=engine)
+
+
+    person_sel = person_tbl.select(person_tbl.c.identifier == person)
+    person = engine.execute(person_sel).fetchone()
+
+    if not person:
+        sys.stderr.write("Can't find anyone in the database with the id {}\n".format(person))
+        sys.exit(1)
+
+    try:
+        img = read_image(image_file)
+    except FileNotFoundError:
+        sys.stderr.write("Can't find the image file {}\n".format(image_file))
+        sys.exit(1)
+
+    conn = engine.connect()
+    conn.execute('pragma foreign_keys=OFF')
+
+    ## Insert the image metadata
+    img_ins = img_tbl.insert()
+    res = conn.execute(img_ins,
+            person_id = person[0],
+            type = image_type,
+            dimensions  = ",".join([str(x) for x in img['dimensions']]),
+            origin  = ",".join([str(x) for x in img['origin']]),
+            spacing = ",".join([str(x) for x in img['spacing']])
+            )
+    image_id = res.inserted_primary_key[0]
+
+    coords = get_coords(img['dimensions'], engine)
+
+    dimensions = img['dimensions']
+    imgdata = img['img']
+    data = []
+    for x in tqdm(range(dimensions[0]), desc="Creating data"):
+        for y in range(dimensions[1]):
+            for z in range(dimensions[2]):
+                cid = coords[x,y,z]
+                value = imgdata[x,y,z]
+                data.append({"img_id": image_id,
+                             "coord_id": cid,
+                             "value": value})
+
+    print("Inserting {} values".format(len(data)))
+    conn.execute(points_tbl.insert(), data)
+
+
+def get_coords(dimensions, engine):
+    coords_tbl = sa.Table('coords', sa.MetaData(), autoload=True, autoload_with=engine)
+
+    add_missing_coords(dimensions, coords_tbl, engine)
+
+    result = engine.execute(coords_tbl.select())
+
+    lookup = np.zeros(dimensions)
+    for i in result:
+        lookup[i[1],i[2],i[3]] = i[0]
+
+    return lookup
+
+
+def add_missing_coords(dimensions, coords_tbl, engine):
```

Comment:
I use the 'EH3041_EH3025_Transformed_Fat.vtk' that I sent you.

It's strange. What version of Python are you running?

Of course, creating ~50 million dictionaries is far from ideal so I was thinking about some alternatives:
- Do we actually need to store the points in the database or can we just use the point id together with the image size to derive the coordinates. Are there any downsides to this? Harder to design queries?
- Another possible solution here is to avoid using dictionaries and just have one separate list for each component (x,y,z). However, I don't know how to do the insert with sqlalchemy using this format.